src/Metadata/Property/Factory/InheritedPropertyNameCollectionFactory.php

```diff
@@ -41,7 +41,7 @@ public function create(string $resourceClass, array $options = []): PropertyName
 
         // Inherited from parent
         foreach ($this->decorated->create($resourceClass, $options) as $propertyName) {
-            $propertyNames[$propertyName] = true;
+            $propertyNames[$propertyName] = (string) $propertyName;
```

Comment:
It looks like the Symfony component suffers from the same bug. We may keep this fix temporarily but we should fix this behavior in Symfony as well then bump the minimal version required by API Platform.

Do you want to handle this? If you're out of time I can open the Symfony PR.